src/GoMath.ts

```diff
@@ -533,149 +539,44 @@ export class GoMath {
-                    white_state += "W" + this.prettyCoords(col, row, width);
-                }
-            }
-        }
-        this_way = `${white_state}.${black_state}`;
-        id = this_way < id ? this_way : id;
-        // console.log("calc, id:", this_way, id);
-
-        black_state = "";
-        white_state = "";
-        for (let row = height - 1; row >= 0; row--) {
-            for (let col = 0; col < width; col++) {
-                if (position[row][col] === JGOFNumericPlayerColor.BLACK) {
-                    black_state += "B" + this.prettyCoords(col, height - row - 1, width);
-                }
-                if (position[row][col] === JGOFNumericPlayerColor.WHITE) {
-                    white_state += "W" + this.prettyCoords(col, height - row - 1, width);
-                }
-            }
-        }
+        // The basic algorithm is to list where each of the stones are, in a long string.
+        // We do this once for each transform, selecting the lowest (lexixally) as we go.
+
+        const tforms: Array<BoardTransform> = [
+            (x, y) => ({ x, y }),
+            (x, y) => ({ x, y: height - y - 1 }),
+            (x, y) => ({ x: y, y: x }),
+            (x, y) => ({ x: y, y: width - x - 1 }),
+            (x, y) => ({ x: height - y - 1, y: x }),
+            (x, y) => ({ x: height - y - 1, y: width - x - 1 }),
+            (x, y) => ({ x: width - x - 1, y }),
+            (x, y) => ({ x: width - x - 1, y: height - y - 1 }),
+        ];
 
-        this_way = `${white_state}.${black_state}`;
-        id = this_way < id ? this_way : id;
-        // console.log("calc, id:", this_way, id);
+        const ids = [];
 
-        black_state = "";
-        white_state = "";
-        for (let row = 0; row < height; row++) {
-            for (let col = width - 1; col >= 0; col--) {
-                if (position[row][col] === JGOFNumericPlayerColor.BLACK) {
-                    black_state += "B" + this.prettyCoords(width - col - 1, row, width);
-                }
-                if (position[row][col] === JGOFNumericPlayerColor.WHITE) {
-                    white_state += "W" + this.prettyCoords(width - col - 1, row, width);
+        for (const tform of tforms) {
+            let black_state = "";
+            let white_state = "";
+            for (let x = 0; x < width; x++) {
+                for (let y = 0; y < height; y++) {
+                    const c = tform(x, y);
+                    if (position[x][y] === JGOFNumericPlayerColor.BLACK) {
+                        black_state += GoMath.encodeMove(c.x, c.y);
+                    }
+                    if (position[x][y] === JGOFNumericPlayerColor.WHITE) {
+                        white_state += GoMath.encodeMove(c.x, c.y);
+                    }
                 }
             }
-        }
+            const this_id =
+                black_state < white_state
+                    ? `${black_state}:${white_state}`
```

Comment:
Note on the colon:

I realize the current use case is not URLs, but `:` isn't as URL-safe as other symbols (`-._~`).  It's not hard to address later (using `replace()` or URL escaping), but I thought it's worth bringing up now in case an unreserved delimiter would be preferable.

Reply:
I wondered why you used `.`   Good reason!

Reply:
I wish I could say why all of `-.,~` feel inferior as a separator to `:`, but OTOH maybe that's why `:` is the separator in URLs 😄